stream2hop/tns2hop.py

```diff
@@ -0,0 +1,254 @@
+    url_tns_api = "https://wis-tns.weizmann.ac.il/api/get"
+
+    #  every 3 hours
+    days_ago = 0.125
+    date = str(datetime.utcnow() - timedelta(days=days_ago))
+    print("Running on: " + date)
+    #  Set search parameter
+    search_obj = {"public_timestamp": date}
+    response = search(url_tns_api, search_obj, api_key)
+
+    if None not in response:
+
+        #  New data is retrieved, open a stream with scimma
+        sC = ut.ScimmaConnection(scimma_url, config)
+        sC.open()
+
+        #  Read parameters file
+        parameters_list = json.load(open("../parameters.json", "r"))["data"]
+        objects_list = json.loads(response.text)["data"]["reply"]
+
+        for object in objects_list:
+            get_obj = {
+                "objname": object["objname"],
+                "photometry": "1",
+                "spectra": "1",
+                "classification": "1",
+                "discovery": "1",
+                "AT": "1",
+            }
+            response = get_object(url_tns_api, get_obj, api_key)
+
+            if response:
+                object_data = json.loads(response.text)["data"]["reply"]
+                object_data = fix_photometery(object_data, parameters_list)
+                object_data = fix_spectra(object_data, parameters_list)
+                object_data = {"format": "BLOB", "content": object_data}
+                sC.write(json.dumps(object_data, indent=4))
+        sC.close()
+    else:
+        print("Nothing to do\n")
+
+    pass
+
+
+def get_astronotes(scimma_url, config, api_key):
+    """
+        Get the latest astronotes in the previous day
+
+        Args:
+            scimma_url: scimma URL that will be used to publish data to it
+            config: configurations to access scimma URL
+
+        Returns:
+            None
+    """
+    date = datetime.today().strftime("%Y-%m-%d")
+    astronotes_url = (
+        "https://wis-tns.weizmann.ac.il/astronotes?&date_start%5Bdate%5D="
+        + date
+        + "&format=csv"
+    )
+    wget.download(astronotes_url, "astronotes_today.csv")
+    with open("astronotes_today.csv") as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=",")
+        line_count = 0
+        headlines = []
+        if len(csv_reader) == 1:
+            print("Nothing to do")
+            pass
+        for row in csv_reader:
+            if line_count == 0:
+                for headline in row:
+                    headlines.append(headline)
+                line_count += 1
+            else:
+                json_object = {"format": "BLOB", "content": {}}
+                for i in range(0, len(headlines)):
+                    json_object["content"][headlines[i]] = row[i]
+                #  New data is retrieved, open a stream with scimma
+                sC = ut.ScimmaConnection(scimma_url, config)
+                sC.open()
+                sC.write(json.dumps(json_object, indent=4))
+                sC.close()
+    os.remove("astronotes_today.csv")
```

Comment:
What is this doing? I'm not able to follow too well.

Reply:
TNS doesn't have an API to retrieve the astronotes, however they have a link to download the astronotes as CSV and you have to play around the URL as filters or query. I used here the date, so I can get all the astronotes in this day in a CSV file.
After I get the astronotes, I keep the field's in the headline in a separate variable to be used as keys in the JSON object. I did this just in case they changed the arrangement of the columns or added new fields. After that, I create the JSON object and pass it to Kafka

Reply:
They have a bug in downloading json format files, but they apparently contain more data than CSV files. When they fix it, we probably will move to JSON files. Maybe they will make APIs later and that will be better than downloading files just for the sake of consistency in the code.

Reply:
Okay, thanks for the explanation, that makes sense.